conn.go

```diff
@@ -215,6 +218,25 @@ func (s *Session) dial(ip net.IP, port int, cfg *ConnConfig, errorHandler ConnEr
 
 	frameTicker := make(chan struct{}, 1)
 	startupErr := make(chan error)
+	// need to call before we write to net.Conn
+	go func() {
+		for {
+			select {
+			case args := <- c.frameWriteArgChan:
+				if err := args.req.writeFrame(args.framer, args.stream); err != nil{
+					// I think this is the correct thing to do, im not entirely sure. It is not
+					// ideal as readers might still get some data, but they probably wont.
+					// Here we need to be careful as the stream is not available and if all
+					// writes just timeout or fail then the pool might use this connection to
+					// send a frame on, with all the streams used up and not returned.
+					c.closeWithError(err)
+				}
```

Comment:
should exit out of this loop on error

Reply:
done, although it should exit `case <-c.quit` even if it is not explicitly returned in the loop.